### ISSUE TITLE
Go Module Version Error

### ISSUE THREAD

**Issue by niallnsec** [coreos/go-systemd#325]
Since the v22 release I can no longer update dependencies in many of my projects. Running go get -u results in the following error:
`go: github.com/coreos/go-systemd@v0.0.0-20191111152658-2d78030078ef: go.mod has post-v0 module path "github.com/coreos/go-systemd/v22" at revision 2d78030078ef`

I firsttried:
 - changing it to use v22.0.0 explicitly
 - specifying +incompatible on the import in go.mod
 - downloading v21 locally and specifying an override
None of that seems to help. Each of these attempts generated a slightly different error message with `go.mod has post-v0 module path "github.com/coreos/go-systemd/v22"` being constant every time.

The only way I was able to get any projects importing the library to build was to download v22.0.0 locally and edit the go.mod file to remove the /v22 suffix from the module import path. Once that is done I added an override to my projects go.mod to replace github.com/coreos/go-systemd with the local copy.

I am not clear as to why this is causing so much trouble or if I am maybe missing something here?

**Comment by lucab** [comment]
Thanks for the report.
I've seen a lot of oddities with Go modules so far, and most of them were related to either bugs in tooling or complex dependency graphs. Before venturing into deeper debugging, I would suggest that you:
 * make sure you are using latest Go toolchain (at this time, 1.13.4)
 * start from a basic example with no dependencies other than go-systemd, make it work, then add back your other dependencies

**Comment by binderclip** [comment]
@lucab Could you please change v22.0.0 to v22?

**Comment by lucab** [comment]
@binderclip unfortunately I cannot, as Go modules require semver tags. Here is an overview on recent releases https://github.com/coreos/go-systemd/issues/320#issuecomment-552499085.

**Comment by codyaray** [comment]
I'm still seeing this same error today. :(

**Comment by forrling** [comment]
Just fixed this for some of my things.
since the go.mod file states `module github.com/coreos/go-systemd/v22` you will need to change your imports

from for example `import "github.com/coreos/go-systemd/journal"`
to `import "github.com/coreos/go-systemd/v22/journal"`

**Comment by nanasi880** [comment]
I have workaround.
Add my go.mod file this.
`replace github.com/coreos/go-systemd => github.com/coreos/go-systemd/v22 latest`

**Comment by lucab** [comment]
This ticket is unfortunately drifting toward being a duplicate of https://github.com/coreos/go-systemd/issues/321.

I've already left all the feedback I can in my original reply here (i.e. check your Go version) and in https://github.com/coreos/go-systemd/issues/321#issuecomment-565367274 (i.e. use versioned imports).

I'm thus going to close and lock this ticket in order to prevent the [Go module fallout reports](https://github.com/coreos/go-systemd/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3Aarea%2Fgomod-fallout+) from sprawling even further. Feel free to bring any new and relevant feedback there.
